terraform/container-app/functions.tf

```diff
@@ -45,4 +49,39 @@ resource "azurerm_linux_function_app" "contentful_function" {
     identity_ids = [azurerm_user_assigned_identity.user_assigned_identity.id]
   }
 
+  app_settings = {
+    AZURE_SQL_CONNECTIONSTRING      = "@Microsoft.KeyVault(VaultName=${azurerm_key_vault.vault.name};SecretName=${azurerm_key_vault_secret.vault_secret_database_connectionstring.name})"
+    AzureWebJobsServiceBus          = azurerm_servicebus_namespace.service_bus.default_primary_connection_string
+    WEBSITE_ENABLE_SYNC_UPDATE_SITE = true
+    WEBSITE_MOUNT_ENABLED           = 1
+    AZURE_CLIENT_ID                 = azurerm_user_assigned_identity.user_assigned_identity.client_id
+    AZURE_KEYVAULT_CLIENTID         = azurerm_user_assigned_identity.user_assigned_identity.client_id
+    AZURE_KEYVAULT_RESOURCEENDPOINT = "https://${local.resource_prefix}-kv.vault.azure.net/"
+    AZURE_KEYVAULT_SCOPE            = "https://vault.azure.net/.default"
+    KeyVaultReferenceIdentity       = data.azurerm_user_assigned_identity.assigned_identity.id
+  }
+}
+
+data "azurerm_function_app_host_keys" "default" {
```

Comment:
I think this will use the function _app_ level keys, but I think we want the individual _function's key_ for better security.

Not sure if this is at all possible?